tests/internal/scripts/mpi/nonblocking.py

```diff
@@ -40,15 +40,12 @@
 # ...
 
 # ...
-before = rank - 1
-after  = rank + 1
+before = np.int32(rank - 1)
+after  = np.int32(rank + 1)
 if rank == 0:
     before = size - 1
 if rank == size - 1:
     after  = 0
-before = np.int32(before)
-after  = np.int32(after )
-# ...
```

Comment:
The original code was ensuring that the variables `before` and `after` passed to `mpi_irecv` are of type `int32`. The new code passes a 32-bit or a 64-bit integer depending on the MPI rank, so it does not look correct to me. What was the reason for this change?

Reply:
the type and precision is defined in the first assignment not the last one

Reply:
Yes but when you do:
```
if rank == 0:
    before = size -1
```
you aren't using a numpy.int32 anymore.  This should raise an error I think (it may be related to issue 325)

Reply:
for the moment we are not strict on the precision and we  assign it from the first assign
so if we do 
```
a = int32(1)
a =  1
```
we still keep the first precision so changing that behaviour is another thing

Reply:
This is why it currently works. However if we added a test on the return type or fix issue 325 then this will no longer be true. It is not good practice to introduce bugs in the test code as we will have errors when we have fixed pyccel's behaviour rather than having errors to show where it is broken

Reply:
okay it's fixed